### PR TITLE
Extend v3io-go API to align with full data model

### PR DESCRIPTION
**Pull request by imakhlin**

### CODE REVIEW

File: pkg/dataplane/http/context.go [hunk 1]
```diff
@@ -163,6 +163,7 @@ func (c *context) GetContainerContents(getContainerContentsInput *v3io.GetContai
 }
 
 // GetContainerContentsSync
+// http://IP:8081/bigdata/?prefix-info=1&prefix-only=1&max-keys=1&marker=naipi_files
```
Comment (gtopper) [inline pkg/dataplane/http/context.go]: What's this comment for?

File: pkg/dataplane/http/context.go [hunk 2]
```diff
@@ -171,6 +172,34 @@ func (c *context) GetContainerContentsSync(getContainerContentsInput *v3io.GetCo
 		query += "prefix=" + getContainerContentsInput.Path
 	}
 
+	if getContainerContentsInput.DirectoriesOnly {
+		if len(query) > 0 {
+			query += "&"
```
Comment (gtopper) [inline pkg/dataplane/http/context.go]: Use `strings.Builder`?

File: pkg/dataplane/http/context.go [hunk 3]
```diff
@@ -294,7 +313,7 @@ func (c *context) GetItemsSync(getItemsInput *v3io.GetItemsInput) (*v3io.Respons
 	c.logger.DebugWithCtx(getItemsInput.Ctx, "Body", "body", string(response.Body()))
 
 	getItemsResponse := struct {
-		Items            []map[string]map[string]interface{}
+		Items []map[string]map[string]interface{}
```
Comment (gtopper) [inline pkg/dataplane/http/context.go]: That looks wrong... Run `go fmt`?

File: pkg/dataplane/test/sync_test.go [hunk 4]
```diff
@@ -34,15 +35,151 @@ func (suite *syncContainerTestSuite) TestGetContainers() {
 	// get containers
 	response, err := suite.container.GetContainersSync(&getContainersInput)
 	suite.Require().NoError(err, "Failed to get containers")
+	defer response.Release()
```
Comment (gtopper) [inline pkg/dataplane/test/sync_test.go]: You don't even have to defer it.

File: pkg/dataplane/http/context.go (hunk 3 above)
Comment (imakhlin) [reply pkg/dataplane/http/context.go]: this is IDE formatter, I'll check

File: pkg/dataplane/test/sync_test.go [hunk 5]
```diff
@@ -34,15 +35,151 @@ func (suite *syncContainerTestSuite) TestGetContainers() {
...
+		Path: path,
+	}
+
+	// when run against a context
+	suite.populateDataPlaneInput(&getContainerContentsInput.DataPlaneInput)
+
+	// get container contents
+	response, err := suite.container.GetContainerContentsSync(&getContainerContentsInput)
+	suite.Require().NoError(err, "Failed to get container contents")
+	defer response.Release()
```
Comment (gtopper) [inline pkg/dataplane/test/sync_test.go]: No need to defer in fact.

File: pkg/dataplane/http/context.go (hunk 3 above)
Comment (imakhlin) [reply pkg/dataplane/http/context.go]: fixed

File: pkg/dataplane/test/sync_test.go [hunk 6]
```diff
@@ -34,15 +35,151 @@ func (suite *syncContainerTestSuite) TestGetContainers() {
...
+	for _, content := range getContainerContentsOutput.Contents {
+		validateContent(suite, &content, len(fileContent), false)
+	}
+
+	for _, prefix := range getContainerContentsOutput.CommonPrefixes {
+		validateCommonPrefix(suite, &prefix, false)
+	}
+
+	suite.Require().Equal(5, len(getContainerContentsOutput.CommonPrefixes))
+	fmt.Println(getContainerContentsOutput)
```
Comment (gtopper) [inline pkg/dataplane/test/sync_test.go]: I don't think we want to print anything.

File: pkg/dataplane/test/sync_test.go [hunk 7]
```diff
@@ -34,15 +35,151 @@ func (suite *syncContainerTestSuite) TestGetContainers() {
...
+
+	getContainerContentsOutput := response.Output.(*v3io.GetContainerContentsOutput)
+	suite.Require().Equal(5, len(getContainerContentsOutput.Contents))
+	suite.Require().Equal(path+"file-4.txt", getContainerContentsOutput.NextMarker)
+	suite.Require().Equal(true, getContainerContentsOutput.IsTruncated)
+
+	for _, content := range getContainerContentsOutput.Contents {
+		validateContent(suite, &content, len(fileContent), true)
+	}
+	fmt.Println(getContainerContentsOutput)
```
Comment (gtopper) [inline pkg/dataplane/test/sync_test.go]: I don't think we want to print anything.

File: pkg/dataplane/test/sync_test.go [hunk 8]
```diff
@@ -34,15 +35,151 @@ func (suite *syncContainerTestSuite) TestGetContainers() {
...
+	getContainerContentsOutput = response.Output.(*v3io.GetContainerContentsOutput)
+	suite.Require().Equal(5, len(getContainerContentsOutput.Contents))
+	suite.Require().Equal(path+"file-9.txt", getContainerContentsOutput.NextMarker)
+	suite.Require().Equal(false, getContainerContentsOutput.IsTruncated)
+
+	for _, content := range getContainerContentsOutput.Contents {
+		validateContent(suite, &content, len(fileContent), true)
+	}
+
+	fmt.Println(getContainerContentsOutput)
```
Comment (gtopper) [inline pkg/dataplane/test/sync_test.go]: I don't think we want to print anything.

File: pkg/dataplane/test/sync_test.go [hunk 9]
```diff
@@ -52,6 +189,15 @@ func (suite *syncContainerTestSuite) TestGetContainerContents() {
...
+	suite.Require().Empty(len(getContainerContentsOutput.Contents))
+	suite.Require().Equal(8, len(getContainerContentsOutput.CommonPrefixes))
+	suite.Require().Equal(path+"dir-9", getContainerContentsOutput.NextMarker)
+	suite.Require().Equal(false, getContainerContentsOutput.IsTruncated)
+
+	for _, prefix := range getContainerContentsOutput.CommonPrefixes {
+		validateCommonPrefix(suite, &prefix, true)
+	}
+
 	fmt.Println(getContainerContentsOutput)
```
Comment (gtopper) [inline pkg/dataplane/test/sync_test.go]: I don't think we want to print anything.

Comment (gtopper) [reply pkg/dataplane/test/sync_test.go]: (I know there was one pre-existing println in there)

File: pkg/dataplane/types.go [hunk 10]
```diff
@@ -67,30 +68,51 @@ type DataPlaneOutput struct {
...
+	// ETag           string   `xml:"ETag"` // Not in use
+	LastModified string `xml:"LastModified"` // Date in format time.RFC3339: "2019-06-02T14:30:39.18Z"
+
+	Mode         *os.FileMode `xml:"Mode"`         // uint32, e.g. 0100664
+	AccessTime   string      `xml:"AccessTime"`   // Date in format time.RFC3339: "2019-06-02T14:30:39.18Z"
+	CreatingTime string      `xml:"CreatingTime"` // Date in format time.RFC3339: "2019-06-02T14:30:39.18Z"
+	GID          string      `xml:"GID"`          // Hexadecimal representation of GID (e.g. "3e8" -> i.e. "0x3e8" == 1000)
+	UID          string      `xml:"UID"`          // Hexadecimal representation of UID (e.g. "3e8" -> i.e. "0x3e8" == 1000)
+	InodeNumber  *uint32      `xml:"InodeNumber"`  // iNode number
+	// InodeGenID   *int         `xml:"InodeGenId"`   // Not in use
```
Comment (gtopper) [inline pkg/dataplane/types.go]: Dead code 😵☠️

File: pkg/dataplane/types.go [hunk 11]
```diff
@@ -67,30 +68,51 @@ type DataPlaneOutput struct {
...
 type Content struct {
 	XMLName        xml.Name `xml:"Contents"`
 	Key            string   `xml:"Key"`
-	Size           int      `xml:"Size"`
-	LastSequenceID int      `xml:"LastSequenceId"`
-	ETag           string   `xml:"ETag"`
-	LastModified   string   `xml:"LastModified"`
+	Size           *int      `xml:"Size"`           // file size in bytes
+	LastSequenceID *int      `xml:"LastSequenceId"` // greater than zero for shard files
+	// ETag           string   `xml:"ETag"` // Not in use
```
Comment (gtopper) [inline pkg/dataplane/types.go]: Dead code 😵☠️

File: pkg/dataplane/types.go [hunk 12]
```diff
@@ -67,30 +68,51 @@ type DataPlaneOutput struct {
...
+	CommonPrefixes xml.Name    `xml:"CommonPrefixes"`
+	Prefix         string      `xml:"Prefix"`       // directory name
+	LastModified   string      `xml:"LastModified"` // Date in format time.RFC3339: "2019-06-02T14:30:39.18Z"
+	AccessTime     string      `xml:"AccessTime"`   // Date in format time.RFC3339: "2019-06-02T14:30:39.18Z"
+	CreatingTime   string      `xml:"CreatingTime"` // Date in format time.RFC3339: "2019-06-02T14:30:39.18Z"
+	Mode           *os.FileMode `xml:"Mode"`         // uint32, e.g. 040775
+	GID            string      `xml:"GID"`          // Hexadecimal representation of GID (e.g. "3e8" -> i.e. "0x3e8" == 1000)
+	UID            string      `xml:"UID"`          // Hexadecimal representation of UID (e.g. "3e8" -> i.e. "0x3e8" == 1000)
+	InodeNumber    *uint32      `xml:"InodeNumber"`  // iNode number
+	// InodeGenID     *int         `xml:"InodeGenId"`   // Not in use
```
Comment (gtopper) [inline pkg/dataplane/types.go]: Dead code 😵☠️

**Pull request merged**